api/handlers.go

```diff
@@ -557,6 +564,122 @@ func (si *ServerImplementation) LookupApplicationByID(ctx echo.Context, applicat
 	})
 }
 
+// LookupApplicationBoxByIDandName returns the value of an application's box
+// (GET /v2/applications/{application-id}/box)
+func (si *ServerImplementation) LookupApplicationBoxByIDandName(ctx echo.Context, applicationID uint64, params generated.LookupApplicationBoxByIDandNameParams) error {
+	if err := si.verifyHandler("LookupApplicationBoxByIDandName", ctx); err != nil {
+		return badRequest(ctx, err.Error())
+	}
+
+	encodedBoxName := params.Name
+	boxNameBytes, err := logic.NewAppCallBytes(encodedBoxName)
+	if err != nil {
+		return badRequest(ctx, err.Error())
+	}
+	boxName, err := boxNameBytes.Raw()
+	if err != nil {
+		return badRequest(ctx, err.Error())
+	}
+
+	q := idb.ApplicationBoxQuery{
+		ApplicationID: applicationID,
+		BoxName:       boxName,
+	}
+	appid, boxes, round, err := si.fetchApplicationBoxes(ctx.Request().Context(), q)
+
+	if err != nil {
+		msg := fmt.Sprintf("%s: round=?=%d, appid=%d, boxName=%s", errFailedLookingUpBoxes, round, applicationID, encodedBoxName)
+		return indexerError(ctx, fmt.Errorf("%s: %w", msg, err))
+	}
+
+	if len(boxes) == 0 {
```

Comment:
query is structured in such a way that:
* no rows back &rarr; no application found
* one row back with null box &rarr; application exists but without the requested box
* one row back with non-null box in the case of successful search

Reply:
Could you return typed errors to make this behavior more explicit? For example the sql package has `sql.ErrNoRows`.

Reply:
Resolving as this has been implemented.